gap/attributes/isomorph.gi

```diff
@@ -327,3 +327,60 @@ function(S)
   UseIsomorphismRelation(H, G);
   return H;
 end);
+
+InstallMethod(IsomorphismSemigroup,
+"for IsStrongSemilatticeOfSemigroups and a Clifford semigroup",
+[IsStrongSemilatticeOfSemigroups, IsSemigroup and IsFinite],
+function(filt, S)
+  local A, idemps, n, D, N, L, classes, idemp, DC, H, map, SSS, i, j;
+  # decomposes a finite Clifford semigroup S into a strong semilattice of
+  # groups and returns an SSS object.
+  if not (IsCliffordSemigroup(S) and IsFinite(S)) then
+    TryNextMethod();
+  fi;
+  # There should be one idempotent per D-class, i.e. per semilattice element
+  # since the semilattice decomposition is by J-classes, and J = D here
+  A      := Semigroup(Idempotents(S));
+  idemps := Elements(A);
+  n      := Size(idemps);
+
+  # create semilattice
+  D := DigraphReflexiveTransitiveReduction(Digraph(NaturalPartialOrder(A)));
+  # currently wrong way round
+  D := DigraphReverse(D);
+  N := OutNeighbours(D);
```

Comment:
You can directly get this via
```suggestion
  N := ReverseNaturalPartialOrder(A);
```
(although I don't think that necessarily (or at all?) contains `x` in `N[x]` for each `x` - if those are actually necessary, you will still want to add them or modify your code below to acts as if they are there).

Reply:
Thanks @wilfwilson, unfortunately I get a "no method found" error when running ReverseNaturalPartialOrder on some inputs - for example:
```
gap> S := Semigroup(Transformation([1, 2, 4, 5, 6, 3, 7, 8]), Transformation([3, 3, 4, 5, 6, 2, 7, 8]), Transformation([1, 2, 5, 3, 6, 8, 4, 4]));
<transformation semigroup of degree 8 with 3 generators>
gap> IsCliffordSemigroup(S);
true
gap> A := Semigroup(Idempotents(S));
<transformation monoid of degree 8 with 3 generators>
gap> NaturalPartialOrder(A);
[ [ 2, 3, 4 ], [ 4 ], [ 4 ], [  ] ]
gap> ReverseNaturalPartialOrder(A);
Error, no method found! For debugging hints type ?Recovery from NoMethodFound
Error, no 2nd choice method found for `ReverseNaturalPartialOrder' on 1 arguments at /Applications/GAP/lib/methsel2.g:250 called from
<function "HANDLE_METHOD_NOT_FOUND">( <arguments> )
 called from read-eval loop at *stdin*:18
type 'quit;' to quit to outer loop
```
Although you are correct that we don't need `x` in `N[x]` since the homomorphisms in this case are the identity, and the strong semilattice constructor is clever enough to fill that in upon creation.

Reply:
Also, I'm realising that taking the reflexive transitive reduction of `D` is redundant, since the SSS constructor effectively reverts this, so I'll remove that part